src/gui/qgsnewdatabasetablenamewidget.cpp

```diff
@@ -0,0 +1,266 @@
+ email                : elpaso at itopen dot it
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include <QTreeWidgetItemIterator>
+
+#include "qgsnewdatabasetablenamewidget.h"
+#include "qgsapplication.h"
+#include "qgsdataitemproviderregistry.h"
+#include "qgsdataitemprovider.h"
+#include "qgsproviderregistry.h"
+#include "qgsprovidermetadata.h"
+
+QgsNewDatabaseTableNameWidget::QgsNewDatabaseTableNameWidget(
+  QgsBrowserGuiModel *browserModel,
+  const QStringList &providersFilter,
+  QWidget *parent )
+  : QWidget( parent )
+{
+
+  // Initalize the browser
+  if ( ! browserModel )
+  {
+    mBrowserModel = new QgsBrowserGuiModel( this );
+    mBrowserModel->initialize();
+  }
+  else
+  {
+    mBrowserModel = browserModel;
+    mBrowserModel->initialize();
+  }
+
+  setupUi( this );
+
+  mValidationResults->setStyleSheet( QStringLiteral( "* { font-weight: bold; color: red; }" ) );
+
+  QStringList hiddenProviders
+  {
+    QStringLiteral( "special:Favorites" ),
+    QStringLiteral( "special:Drives" ),
+    QStringLiteral( "special:Volumes" ),
+    QStringLiteral( "special:Home" ),
+    QStringLiteral( "special:ProjectHome" )
+  };
+
+  const auto providerList { QgsApplication::dataItemProviderRegistry()->providers() };
+  for ( const auto &provider : providerList )
+  {
+    if ( provider->dataProviderKey().isEmpty() )
+    {
+      hiddenProviders.push_back( provider->name() );
+      continue;
+    }
+    QgsProviderMetadata *md { QgsProviderRegistry::instance()->providerMetadata( provider->dataProviderKey() ) };
+    if ( ! md )
+    {
+      hiddenProviders.push_back( provider->name() );
+      continue;
+    }
+    if ( provider->capabilities() & QgsDataProvider::DataCapability::Database )
+    {
+      if ( ! providersFilter.isEmpty() && ! providersFilter.contains( provider->dataProviderKey() ) )
+      {
+        hiddenProviders.push_back( provider->name() );
+      }
+      else
+      {
+        mShownProviders.insert( provider->dataProviderKey() );
+      }
+    }
+    else
+    {
+      hiddenProviders.push_back( provider->name() );
+    }
+  }
+
+  mBrowserProxyModel.setBrowserModel( mBrowserModel );
+  mBrowserProxyModel.setDataItemProviderKeyFilter( hiddenProviders );
+  mBrowserProxyModel.setShowLayers( false );
+  mBrowserTreeView->setHeaderHidden( true );
+  mBrowserTreeView->setExpandsOnDoubleClick( false );
```

Comment:
Why this?

Reply:
shameless copied from the change datasource dialog.

> Looking good (+nice clean ups!)
> 
> The data source select dialog includes a refresh button. Do you think we should add one here? I'm imagining a use case where someone is asked for a table name, realises they need to create a new schema, and then needs to refresh the tree to show this newly created schema. (Or does the view automatically expose the browser right click actions like refresh and create schema already?)

No, the view has no context menus. What would you prefer: add the context menus or the refresh button, or both?

Reply:
I'm unsure - the context menus sound useful (and less obtrusive) but I'm 50/50 on whether having actions which edit things are good practice in an "open" type dialog. Your call!

Reply:
Added refresh tool button